packages/build-utils/src/fs/download.ts

```diff
@@ -42,14 +42,23 @@ async function prepareSymlinkTarget(
   );
 }
 
-async function downloadFile(file: File, fsPath: string): Promise<FileFsRef> {
+export async function downloadFile(
+  file: File,
+  fsPath: string
+): Promise<FileFsRef> {
   const { mode } = file;
 
+  // if the source is a symlink, try to create it instead of copying the file
+  // however, if the symlink creation fails (e.g. Windows), fallback to copying
   if (isSymbolicLink(mode)) {
     const target = await prepareSymlinkTarget(file, fsPath);
 
-    await symlink(target, fsPath);
-    return FileFsRef.fromFsPath({ mode, fsPath });
+    try {
+      await symlink(target, fsPath);
+      return FileFsRef.fromFsPath({ mode, fsPath });
+    } catch (e) {
+      // fall through
```

Comment:
This fall through case is a bit tricky because for non-`FileFsRef` types (`FileBlob`, specifically), the "contents" of the file are actually the symlink target. So do do this properly we'd need to pass in the full `files` object to this function and replace the `file` reference here to be the resolved one (`file = files[target]` or similar).

Reply:
OK, so what you're saying is if `file` is a `FileBlob` or `FileRef`, the contents IS a symlink and if it fails, we really don't want to fall through.

I'm just worried because a month ago I ran into a problem with creating symlinks on Windows where it will fail with `EPERM` if not running as administrator.

If we truly want symlink creation failures to throw, then perhaps in the `catch` we say if on Windows and code equals `EPERM`, then rewrite the error to say `"Unable to create symlink due to Windows group policy setting; either allow symlinks or run as administrator"`.

Not sure I'm following your reference to `files`. I'm calling `downloadFile()`, not `download()`.

Reply:
I removed the try/catch. We can improve the error should it become an issue.